helpers/limit/limits.go

```diff
@@ -9,16 +9,21 @@ const (
 	goroutinesPerGB float64 = 250000
 )
 
+type Rlimit struct {
+	Cur uint64
+	Max uint64
+}
+
 func GetMaxGoRoutines() uint64 {
 	limit := calculateGoRoutines(getMemory())
-	ulimit, err := getUlimit()
-	if err != nil || ulimit == 0 {
+	ulimit, err := GetUlimit()
+	if err != nil || ulimit.Max == 0 {
 		return limit
 	}
-	if ulimit > limit {
+	if ulimit.Max > limit {
 		return limit
 	}
-	return ulimit
+	return ulimit.Max
```

Comment:
On a seperate/ related note, instead of going all the way up to the `ulimit` why not do something like `ulimit * 0.8` So then we don't try and take all of the descriptors because we know there are other things that use file descriptors other than our go routines

Reply:
We can do that, will do in a separate pr